jolie/src/main/java/jolie/process/InitDefinitionProcess.java

```diff
@@ -31,15 +33,54 @@
 import jolie.runtime.embedding.EmbeddedServiceLoadingException;
 
 public class InitDefinitionProcess extends DefinitionProcess {
+
+	/**
+	 * A `CompletableFuture` that holds the result of the execution.
+	 *
+	 * It is completed with either `null` if the initialization of the service is successful or an
+	 * `Exception` if an error occurs.
+	 */
+	private final CompletableFuture< Exception > executionDone;
```

Comment:
Mmmh it's a bit weird to have this in a node of the interpretation tree. They are supposed to be immutable/as stateless as possible.

Maybe we could put the future in the `Interpreter` class?